src/torchmetrics/functional/classification/stat_scores.py

```diff
@@ -416,6 +416,8 @@ def _multiclass_stat_scores_update(
         fp = confmat.sum(0) - tp
         fn = confmat.sum(1) - tp
         tn = confmat.sum() - (fp + fn + tp)
+        if ignore_index is not None:
+            fp[ignore_index] = 0
```

Comment:
this is probably correct, but since many metrics derive from the `stat_scores` class that means that basically all would need to have their unittests fixed

Reply:
what do you mean by it needs to be fixed?